build/package/install_agent_spawn_host.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+set -ex
+
+while getopts 'i:h:t:u:v:' opt; do
+    case $opt in
+        i) keyfile="$OPTARG";;      # SSH identity file
+        u) user="$OPTARG";;         # Username on the remote host
+        h) hostsfile="$OPTARG";;    # Output of Evergreen host.list
+        t) groupid="$OPTARG";;      # APIGroupId
+        v) apiKey="$OPTARG";;       # APIKey
+        *) exit 1;;
+    esac
+done
+
+hosts=$(cat << EOF | python - "$hostsfile"
+import sys
+import json
+with open(sys.argv[1]) as hostsfile:
+    hosts = json.load(hostsfile)
+    for host in hosts:
+        print(host["dns_name"])
+EOF
+)
+for host in $hosts; do
+    set +e
+    echo "Installing dependeces on $host"
+    ssh -i "$keyfile" -o ConnectTimeout=50 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -tt "$user@$host" 'bash -s' <<'ENDSSH'
+        free -m
+        sudo apt-get install -y --no-install-recommends ca-certificates curl logrotate openssl snmp && exit
+ENDSSH
+    echo "Installing the automation agent on $host"
+    ssh -i "$keyfile" -o ConnectTimeout=50 -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -tt "$user@$host" ARG1="$groupid" ARG2="$apiKey" 'bash -s' <<'ENDSSH'
+        free -m
+        echo "Downloadind and extracting the automation agent"
+        curl -OL https://cloud-dev.mongodb.com/download/agent/automation/mongodb-mms-automation-agent-manager_10.15.0.6409-1_amd64.ubuntu1604.deb
```

Comment:
[nit] two things here, let's get the agent from cloud (https://cloud.mongodb.com) and can we use latest here? as in 
`mongodb-mms-automation-agent-manager_latest`? I think 
`https://cloud.mongodb.com/download/agent/automation/mongodb-mms-automation-agent-manager_latest_amd64.ubuntu1604.deb` works
The main reason to use latest is os we don't have to go updating this test and the reason to use cloud is to use an stable release

Reply:
After making this change one cloud manager e2e test is failing and the agent is no longer display in the UI
```
Invalid config: You need to upgrade the automation agent before publishing other changes
[2020/05/28 17:52:37.759]     --- FAIL: TestCloudManagerDBUsers/Create (0.20s)
```

Reply:
I found the issue: since we are downloading the agent from **cloud.mongodb.com** instead of **cloud-dev.mongodb.com**, the property `mmsBaseUrl` was pointing to **cloud.mongodb.com**. Unfortunately, the following piece of code does not work if we have special characters.

```
sudo sed -i "s/\(mmsGroupId *= *\).*/\1$ARG1/" /etc/mongodb-mms/automation-agent.config
sudo sed -i "s/\(mmsApiKey *= *\).*/\1$ARG2/" /etc/mongodb-mms/automation-agent.config
```


**I fixed this issue by downloading the latest from cloud-dev**. I don't think that makes a difference here since the main point was to download always the latest version of the automation agent. Let me know if this solution is okay with you @gssbzn